avm/utilities/pipelines/platform/Invoke-AvmJsonModuleIndexGeneration.ps1

```diff
@@ -154,16 +179,44 @@ function Invoke-AvmJsonModuleIndexGeneration {
     $lastModuleIndexData = $lastModuleIndexJsonFileContent | ConvertFrom-Json -Depth 10
     $currentGeneratedModuleIndexData = $currentGeneratedModuleIndexJsonFileContent | ConvertFrom-Json -Depth 10
 
-    $mergedModuleIndexData = $lastModuleIndexData + $currentGeneratedModuleIndexData | Group-Object -Property moduleName | ForEach-Object {
-      $_.Group | Sort-Object -Property @{Expression = { $_.tags -join ',' } } -Descending | Select-Object -First 1
-    } | Sort-Object -Property moduleName
+    $initialMergeOfJsonFilesData = @{}
+
+    foreach ($module in $currentGeneratedModuleIndexData) {
+      $initialMergeOfJsonFilesData[$module.moduleName] = $module
+    }
+
+    # Add modules from lastModuleIndexData to the initialMergeOfJsonFilesData hashtable, merging tags and properties if they exist in both files
+    foreach ($module in $lastModuleIndexData) {
+      if (-not $initialMergeOfJsonFilesData.ContainsKey($module.moduleName)) {
+        $initialMergeOfJsonFilesData[$module.moduleName] = $module
+      } else {
+        # If the module exists, merge the tags and properties
+        $mergedModule = $initialMergeOfJsonFilesData[$module.moduleName]
+        $mergedModule.tags = @(($mergedModule.tags + $module.tags) | Sort-Object | Get-Unique)
+
+        # Merge properties
+        foreach ($property in $module.properties.PSObject.Properties) {
+          if (-not $mergedModule.properties.PSObject.Properties.Name.Contains($property.Name)) {
+            $mergedModule.properties | Add-Member -NotePropertyName $property.Name -NotePropertyValue $property.Value
+          }
+        }
+      }
+    }
+
+    # Convert the mergedModuleIndexData hashtable to an array of values (i.e., the modules)
+    $mergedModuleIndexData = $initialMergeOfJsonFilesData.Values
+
+    # Sort the modules by their names
+    $sortedMergedModuleIndexData = $mergedModuleIndexData | Sort-Object moduleName
 
     Write-Verbose "Convert mergedModuleIndexData variable to JSON and save as 'moduleIndex.json'" -Verbose
-    $mergedModuleIndexData | ConvertTo-Json -Depth 10 | Out-File -FilePath $moduleIndexJsonFilePath
+    $sortedMergedModuleIndexData | ConvertTo-Json -Depth 10 | Out-File -FilePath $moduleIndexJsonFilePath
   }
   if ($doNotMergeWithLastModuleIndexJsonFileVersion -eq $true) {
-    Write-Verbose "Convert currentGeneratedModuleIndexData variable to JSON and save as 'moduleIndex.json to overwrite it as doNotMergeWithLastModuleIndexJsonFileVersion was set to true'" -Verbose
+    Write-Verbose "Convert currentGeneratedModuleIndexData variable to JSON and save as 'moduleIndex.json to overwrite it as `doNotMergeWithLastModuleIndexJsonFileVersion` was specified'" -Verbose
     $moduleIndexData | ConvertTo-Json -Depth 10 | Out-File -FilePath $moduleIndexJsonFilePath -Force
   }
 
+  Write-Output ('{0}={1}' -f 'anyErrorsOccurred', $anyErrorsOccurred) >> $env:GITHUB_ENV
```

Comment:
As per our practices, I'd recommend to change this to a `return ($anyErrorsOccurred ? $false : $true)`

and in the pipeline change the invocation to
```yaml
if(-not (Set-AzStorageBlobContent @functionInput -Force) {
  Write-Output ('{0}={1}' -f 'anyErrorsOccurred', $true) >> $env:GITHUB_ENV
}
```

Reply:
Background is that we try to keep environment-specific code out of the scripts. To be fair, in the CARML days this practice developed becauase we wanted to be CI/CD-tool-agnostic. 😏